### PR TITLE
[RDBMS] Breaking Change: Parameter name updates. Improvements to Management Plane for MySQL and PostgreSQL

### PR DESCRIPTION
**Pull request by mjain2**
**Description**  

* Note: Recreating this PR to update the history/rebase with hotfix changes for rdbms.

This PR improves existing commands for the 'Flexible Servers' APIs within the RDBMS component.  

Breaking changes:
_Parameter name changes_
1. -- time to --restore-in-time for restore command
Why: PM requested change to make it clearer for customer.  Documentation on our side will also be updated. 
2. --force to --yes 
Why: Consistency across different delete prompt commands for customer experience.
3. --server-name to --name for CRUD operations
Why: PM requested change for consistency
4. --name to --rule-name in firewall rule
Why: PM requested for clearer & self-explanatory params for customer experience.

Changes include:

- Update MySQL Connection string in create command
- Replace server name check with CheckNameAvailability API for MySQL & PostgreSQL
- Update help text and parameters for commands to be more descriptive and helpful to user (commands/arguments: vnet related, replica, firewall, restore).  These were to improve customer experience.
- Update validators to use result from list-skus commands/API.  
- Enabled more tests for code coverage and command testing (validator & replica tests)
- Added new recording files with default locations for MySQL & PostgreSQL 

**Testing Guide**  

To run tests (these cover existing commands and include new tests):
azdev test rdbms

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by mjain2** [comment]
Please merge this with a squash commit :)

**Review by DaeunYim — APPROVED** [review]

**Comment by yonzhan** [comment]
RDBMS

**Comment by mjain2** [comment]
Reverted last two accidental commits.

File: src/azure-cli/azure/cli/command_modules/rdbms/_helptext_mysql.py [hunk 1]
```diff
@@ -16,6 +16,15 @@
 type: command
 short-summary: Create a flexible server.
 examples:
+  - name: Create a MySQL flexible server with default params (resource group, location, servername, username, password) with VNET enabled by default.
+    text: |
+        az mysql flexible-server create
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/rdbms/_helptext_mysql.py:21]: no parameter is required for this command?

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py [hunk 2]
```diff
@@ -278,16 +278,20 @@ def _flexible_server_params(command_group):
 
         with self.argument_context('{} flexible-server delete'.format(command_group)) as c:
             c.argument('server_name', id_part='name', options_list=['--name', '-n'], arg_type=server_name_getter_arg_type)
-            c.argument('force', options_list=['--force'], action='store_true',
-                       help='Delete the server without prompt')
+            c.argument('yes', options_list=['--yes', '-y'], action='store_true',
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/rdbms/_params.py:281]: renaming parameters name is breaking change. could you explain why we need such renaming?

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py [hunk 3]
```diff
@@ -278,16 +278,20 @@ def _flexible_server_params(command_group):
...
+                       help='Do not prompt for confirmation.')
 
         with self.argument_context('{} flexible-server restore'.format(command_group)) as c:
             c.argument('server_name', options_list=['--name', '-n'], arg_type=overriding_none_arg_type,
                        help='The name of the new server that is created by the restore command.')
-            c.argument('source_server', options_list=['--source-server'],
-                       help='The name or resource ID of the source server to restore from.')
-            c.argument('restore_point_in_time', options_list=['--time'],
+            c.argument('restore_point_in_time', options_list=['--restore-time'],
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/rdbms/_params.py:287]: same as above.

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py [hunk 4]
```diff
@@ -374,7 +378,7 @@ def _flexible_server_params(command_group):
                        help='The start IP address of the firewall rule. Must be IPv4 format. Use value \'0.0.0.0\' to represent all Azure-internal IP addresses. ')
 
         with self.argument_context('{} flexible-server firewall-rule delete'.format(command_group)) as c:
-            c.argument('prompt', options_list=['--prompt'], help='Turn confirmation prompt on/off. If off, the rule will be deleted without confirmation')
+            c.argument('yes', options_list=['--yes', '-y'], action='store_true', help='Do not prompt for confirmation.')
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/rdbms/_params.py:381]: same

File: src/azure-cli/azure/cli/command_modules/rdbms/_helptext_mysql.py (hunk 1 above)
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/_helptext_mysql.py:21]: Yes, we want to make creating step simple and easy so we auto-generate everything if the user doesn't pass along any parameters

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py (hunk 2 above)
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/_params.py:281]: we wanted to deleting align confirmation parameters same as deleting commands of other resources

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py (hunk 3 above)
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/_params.py:287]: PMs decided to go with restore-time. Originally it was --restore-point-in-time but we needed to fix this due to linter issue and we are going to stick to this parameter name from now on..

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py (hunk 4 above)
Comment (DaeunYim) [reply src/azure-cli/azure/cli/command_modules/rdbms/_params.py:381]: we wanted to deleting align confirmation parameters same as deleting commands of other resources

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py (hunk 2 above)
Comment (mjain2) [reply src/azure-cli/azure/cli/command_modules/rdbms/_params.py:281]: @Juliehzl have updated the PR to reflect these changes.  Our new service was just rolled out last week so usage is still low, so there is relatively low risk.

File: src/azure-cli/azure/cli/command_modules/rdbms/_params.py (hunk 4 above)
Comment (mjain2) [reply src/azure-cli/azure/cli/command_modules/rdbms/_params.py:381]: See above (and PR description udpates)

**Comment by Juliehzl** [comment]
could you refine description part according to https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests?

**Comment by mjain2** [comment]
> could you refine description part according to https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests?

Made the update thanks!

**Review by Juliehzl — APPROVED** [review]
Approve for the breaking change because there is little usages in past 30 days and service team request

**Pull request merged**